sim/paladin/retribution/retribution.go

```diff
@@ -0,0 +1,79 @@
+package retribution
+
+import (
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+	"github.com/wowsims/sod/sim/paladin"
+)
+
+func RegisterRetributionPaladin() {
+	core.RegisterAgentFactory(
+		proto.Player_RetributionPaladin{},
+		proto.Spec_SpecRetributionPaladin,
+		func(character *core.Character, options *proto.Player) core.Agent {
+			return NewRetributionPaladin(character, options)
+		},
+		func(player *proto.Player, spec interface{}) {
+			playerSpec, ok := spec.(*proto.Player_RetributionPaladin) // I don't really understand this line
+			if !ok {
+				panic("Invalid spec value for Retribution Paladin!")
+			}
+			player.Spec = playerSpec
+		},
+	)
+}
+
+func NewRetributionPaladin(character *core.Character, options *proto.Player) *RetributionPaladin {
+	retOptions := options.GetRetributionPaladin()
+
+	pal := paladin.NewPaladin(character, options.TalentsString)
+
+	ret := &RetributionPaladin{
+		Paladin:     pal,
+		PrimarySeal: retOptions.Options.PrimarySeal,
+	}
+
+	// ret.PaladinAura = retOptions.Options.Aura
+
+	ret.EnableAutoAttacks(ret, core.AutoAttackOptions{
+		MainHand:       ret.WeaponFromMainHand(0), // Set crit multiplier later when we have targets.
```

Comment:
You can set Crit Multiplier alraedy as it does not depend on target

Reply:
Ah, I see this interacts with the above feedback. Given the crit multiplier on melee swings is constant between prot and ret, should it be set in paladin.go like we describe above, or in the individual retribution.go and eventually protection.go files?

Reply:
I think we keep auto attack initialization in the specific specs

Reply:
handled in https://github.com/wowsims/sod/pull/302/commits/eeaee60237a3194c07a710dc6ac986c583373ef8